dot/network/notifications.go

```diff
@@ -128,8 +133,13 @@ func (s *Service) createNotificationsMessageHandler(
 			ok   bool
 			msg  NotificationsMessage
 			peer = stream.Conn().RemotePeer()
+			hs   Handshake
 		)
 
+		if hs, ok = m.(Handshake); ok && hs.IsHandshake() {
```

Comment:
That feels weird to do an interface assertion and then check if it's a handshake again.

We should do one or the other, not both

Reply:
Yea do we still need `IsHandshake()` if we can do the type assertions?

Reply:
> That feels weird to do an interface assertion and then check if it's a handshake again.

`IsHandshake` not only checks the type but checks if it is a valid handshake. I am going to rename `IsHandshake` to `IsValidHandshake`, that would make things more clear.